pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -83,7 +87,14 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			log.Error(err, "failed to get NSX resource path from subnet", "subnetport", subnetPort)
 			return common.ResultRequeue, err
 		}
-		nsxSubnetPortState, err := r.Service.CreateOrUpdateSubnetPort(subnetPort, nsxSubnetPath, "")
+		labels, err := r.getLabelsFromVirtualMachine(ctx, subnetPort)
+		if err != nil {
+			log.Error(err, "failed to get labels from virtualmachine", "subnetPort.Name", subnetPort.Name, "subnetPort.UID", subnetPort.UID, "subnetPort.Spec.AttachmentRef", subnetPort.Spec.AttachmentRef)
```

Comment:
if err != nil, does the logic keep handling or need to reenqueue?
and if err != nil, no need to check if labels != nil

Reply:
Good point! I always forgot to return the non-nil `err`..